app.go

```diff
@@ -163,6 +172,22 @@ func (s *StoreState) SetFeature(name string, val bool) {
 	s.store.Set(featureKey(name), data)
 }
 
+func (s *StoreState) ChangeConfigSetting(name, value string) error {
+	if err := config.SetConfigSetting(s.config, name, value); err != nil {
+		return err
+	}
+	configBytes, err := proto.Marshal(s.config)
+	if err != nil {
+		return err
+	}
```

Comment:
If the marshalling fails the config change should be rolled back, otherwise replay will break.